### DISCUSSION TITLE
Can Mediamtx Provide Channel Numbers?

### DISCUSSION THREAD

**Discussion by PK232** [bluenviron/mediamtx#3832] [Questions]
### Question

I downloaded the current mediamtx binary and installed it on a Linux server (Debian) that has bookworm and is current with all updates. I next configured mediamtx.yml to source my IP cameras. Right out of the gate, I could access the RSTP stream of each camera via the server using VLC.  It was a pleasant surprise and I thought it was going to be downhill from there.  Unfortunately that was not the case. As a result I have come here after failing to find a solution on my own. I suspect however, based on what little information I could find, that instead of a solution, there will only be an explanation of why there isn't one. Time will tell.

A typical URL that I used successfully with VLC is

rtsp://serverIP:8554/camera_name

The problem arises because the software on my always on tablets that I currently use for monitoring the cameras directly is configured with separate spots for camera IP addresss, port number, and channel number. I tried using the stream name for the channel number in the tablet software and that did not work.  I next tried to change the camera name to a two digit channel number in mediamtxc.yml and that did not work as well.  I received an error message indicating that "integer keys are not supported".  

One possible solution would be if the server could provide channel numbers instead of stream names, but I could find nothing in the instructions on how to accomplish that, and a search of past discussions using “channel” as a search term also turned up nothing useful. A wider search on the web, was not helpful as well. I suspect the problem is that I want mediamtx to provide apples when it is built to only provide oranges, but perhaps there is a chance I am wrong, hence this posting.  Thanks.

**Comment by aler9** [comment]
Hello, if you want to use numbers as keys in YAML, it's enough to put them inside quotes:

```yml
paths:
  '1':
    source: rtsp://my-cam-1
  '2':
    source: rtsp://my-cam-2
```
